### PR TITLE
Fix/correct cudf installed package requirements

### PR DESCRIPTION
**Pull request by robertmaynard**
This rolls up all the cudf changes from https://github.com/rapidsai/cudf/pull/7674 that are necessary for `libcudf_kafka` to build against an installed version of `branch-0.19`.

### CODE REVIEW

File: cpp/cmake/cudf-config.cmake.in [hunk 1]
```diff
@@ -1,7 +1,68 @@
...
+targets:
+
+ cudf::cudf             - The main cudf library.
+
+This module offers an optional testing component which defines the
+following IMPORTED GLOBAL  targets:
+
+ cudf::cudftestutil     - The main cudf testing library
+ cudf::gmock_main
+ cudf::gtest_main
```
Comment (kkraus14) [inline cpp/cmake/cudf-config.cmake.in:30]: are `cudf::gmock` and `cudf::gtest` not `IMPORTED GLOBAL` targets as well?

File: cpp/cmake/cudf-config.cmake.in [hunk 2]
```diff
@@ -1,7 +1,68 @@
...
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+If cudf is found, this module defines the following IMPORTED GLOBAL
+targets:
+
+ cudf::cudf             - The main cudf library.
+
+This module offers an optional testing component which defines the
+following IMPORTED GLOBAL  targets:
```
Comment (kkraus14) [inline cpp/cmake/cudf-config.cmake.in:26]:
```suggestion
following IMPORTED GLOBAL targets:
```

Comment (robertmaynard) [reply cpp/cmake/cudf-config.cmake.in:26]: They are, I didn't know if we wanted to formally document them.

File: cpp/cmake/cudf-config.cmake.in (hunk 1 above)
Comment (robertmaynard) [reply cpp/cmake/cudf-config.cmake.in:30]: They are, I didn't know if we wanted to formally document them.

Comment (kkraus14) [reply cpp/cmake/cudf-config.cmake.in:30]: I figure we should either document them all, or only document the `cudf::cudftestutil`

Comment (robertmaynard) [reply cpp/cmake/cudf-config.cmake.in:30]: I will document them all

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**